e2e/main_test.go

```diff
@@ -99,9 +119,108 @@ func setupCluster(ctx context.Context, t testing.TB) (client.Client, *rest.Confi
 		return nil, nil, err
 	}
 	t.Log(">>> operator started successfully")
+
+	if explicitCredentialsConfigured() {
+		t.Log(">>> configure operator explicit credentials")
+		if err := configureOperatorExplicitCredentials(ctx, kubeClient); err != nil {
+			return nil, nil, err
+		}
+	}
 	return kubeClient, restConfig, nil
 }
 
+// Setup explicit credentials for GCM_SECRET or explicit gcpServiceAccount flag.
+// Otherwise, GOOGLE_APPLICATION_CREDENTIALS will use default flow.
+func explicitCredentialsConfigured() bool {
+	return !skipGCM && (gcpServiceAccount != "" || os.Getenv(gcmSecretEnv) != "")
+}
+
+func getExplicitGCMSAJSON() ([]byte, error) {
+	if gcpServiceAccount != "" {
+		b, err := os.ReadFile(gcpServiceAccount)
+		if err != nil {
+			return b, fmt.Errorf("read service account file %q: %w", gcpServiceAccount, err)
+		}
+		return b, nil
+	}
+	if gcmSA := os.Getenv(gcmSecretEnv); gcmSA != "" {
+		return []byte(gcmSA), nil
+	}
+	return nil, errors.New("gcp-service-account flag or GCM_SECRET environment variable not set")
+}
+
+func createGCMSecret(ctx context.Context, kubeClient client.Client, serviceAccount []byte) error {
+	if err := kubeClient.Create(ctx, &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gcmExplicitSecretName,
+			Namespace: operator.DefaultPublicNamespace,
+		},
+		Data: map[string][]byte{
+			gcmExplicitSecretKey: serviceAccount,
+		},
+	}); err != nil {
+		return fmt.Errorf("create GCM service account secret: %w", err)
+	}
+	return nil
+}
+
+// collectorExplicitCredentials returns the local path for operator injected
+// credentials in the local collector filesystem, when using createGCMSecret().
+func collectorExplicitCredentials() string {
+	return fmt.Sprintf("/etc/secrets/secret_%s_%s_%s", operator.DefaultPublicNamespace, gcmExplicitSecretName, gcmExplicitSecretKey)
+}
+
+// newMetricClient returns GCM client. Our e2e tests supports both GOOGLE_APPLICATION_CREDENTIALS (file)
+// and GCM_SECRET (content, required with CI use).
+func newMetricClient(ctx context.Context) (*gcm.MetricClient, error) {
+	if !explicitCredentialsConfigured() {
+		return gcm.NewMetricClient(ctx)
+	}
+	gcmSA, err := getExplicitGCMSAJSON()
+	if err != nil {
+		return nil, err
+	}
+	return gcm.NewMetricClient(ctx, option.WithCredentialsJSON(gcmSA))
```

Comment:
If we can mount the contents of the service account as a file, we can avoid introducing `getExplicitGCMSAJSON()` and rely on `option.WithCredentialsFile` here instead.